portingdb/htmlreport.py

```diff
@@ -588,6 +589,82 @@ def ascending(p):
     )
 
 
+def mispackaged():
+    # Parameters
+    requested = request.args.get('requested', None)
+    if requested not in ('1', None):
+        abort(400)  # Bad request
+
+    # Database
+    db = current_app.config['DB']()
+    query = queries.packages(db)
+
+    # Get data
+    mispackaged, query = queries.split(query, tables.Package.status == 'mispackaged')
+    mispackaged = mispackaged.options(subqueryload('collection_packages'))
+    mispackaged = mispackaged.options(subqueryload('collection_packages.links'))
+    mispackaged = mispackaged.options(subqueryload('collection_packages.tracking_bugs'))
+
+    if requested:
+        # Filter only to packages where maintainer requested a patch
+        mispackaged = [pkg for pkg in mispackaged
+                if "https://bugzilla.redhat.com/show_bug.cgi?id=1333765"
+                in pkg.list_tracking_bugs]
+
+    # Sorting the packages based on their last_link_update
+    #   Note that sorting through the database is problematic, because
+    #   to sort on a @property you need to create a @hybrid_property which
+    #   needs to be written as sort of an SQL expression
+
+    # Packages without a BZ report will be last
+    nonevalue = datetime.datetime.utcnow()
+    mispackaged = sorted(mispackaged,
+            key=lambda pkg: pkg.last_link_update or nonevalue)
+
+    # Render the page, pass the data
+    return render_template(
+        'mispackaged.html',
+        breadcrumbs=(
+            (url_for('hello'), 'Python 3 Porting Database'),
+            (url_for('mispackaged'), 'Mispackaged'),
+        ),
+        requested=bool(requested),
+        mispackaged=mispackaged,
+        ago=ago,
```

Comment:
It would be a bit cleaner to use `ago` as a filter, see `app.jinja_env.filters` below.